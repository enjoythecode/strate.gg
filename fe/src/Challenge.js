import { makeObservable, observable, action, computed } from "mobx"
import {Amazons, AmazonsView} from "./Games/Amazons"
import {Mancala, MancalaView} from "./Games/Mancala"
import { observer } from "mobx-react"
import RootState  from "./State.js"
import React from "react"

const game_name_to_state_class = {
    "amazons": Amazons,
    "mancala": Mancala
}

const game_name_to_view_component = {
    "amazons": AmazonsView,
    "mancala": MancalaView
}


class Challenge {
    game_name = ""
    game_state = null
    players = []
    status = ""
    cid = ""
    moves = []
    
    constructor(data){
        makeObservable(this, {
            game_state: observable,
            players: observable,
            status: observable,
            update_challenge_information: action,
            client_turn: computed
        })
        this.cid = data.cid
        this.game_name = data.game_name
        this.game_state = new game_name_to_state_class[this.game_name](this, data.config)
        this.ViewComponent = game_name_to_view_component[this.game_name]
    }

    send_move(move){
        RootState.socket.send_move({"cid": this.cid, "move": move})
    }

    process_new_move(move){
        this.game_state.process_new_move(move)
        this.moves.push(move)
    }

    update_challenge_information(data){
        if("status" in data){this.status = data.status}
        if("players" in data){this.players = data.players}
        if("cid" in data){this.cid = data.cid}
    }

    get client_turn(){ // 0-indexed player turn (ie. 1st player = 0, 2nd player = 1, ...), -1 if not a player
        return this.players.findIndex((player) => player === RootState.socket.socket_id)
    }
}

const PlayerTagView = (props) => {

    let classes = ["player-tag"]
    if (props.isSelf){
        classes.push("player-tag-self")
    }

    let display;
    if(typeof props.displayName !== 'undefined'){
        console.log("dn", props.displayName)
        display = "Anon#" + props.displayName.slice(0, 7)
    }else{
        display = "Waiting for opponent..."
    }

    return (
        <div>
            <span className={classes.join(" ")}>{display}</span>
        </div>
    )
}


const ChallengeView = observer(({ challenge }) =>(
    <div>
        {
        challenge == null ? "Loading the game!" :
            <div>
                <div className="challenge-wrapper">
                    <div className="challenge-board">
                        <challenge.ViewComponent game_state={challenge.game_state}/>
                    </div>
                    <div className="challenge-dashboard">
                        <div>
                            <PlayerTagView displayName={challenge.players[1]} isSelf={challenge.client_turn == 1}/>
                            <h3>Move List</h3>
                            <h4>Status: {challenge.status}</h4>
                            <PlayerTagView displayName={challenge.players[0]} isSelf={challenge.client_turn == 0}/>
                        </div>
                    </div>
                </div>
            </div>
        }
    </div>
))

export {Challenge, ChallengeView}