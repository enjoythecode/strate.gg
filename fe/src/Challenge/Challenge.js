import { makeObservable, observable, action, computed } from "mobx";
import { AmazonsLogic } from "Games/Amazons/AmazonsLogic";
import { AmazonsView } from "Games/Amazons/AmazonsView";

const game_name_to_state_class = {
  amazons: AmazonsLogic,
};

const game_name_to_view_component = {
  amazons: AmazonsView,
};

/*
A challenge is represented as a Dictionary with the following structure:
{
    "game_name": String. Representing the type of game this challenge is playing
        ex: "amazons" or "mancala",
    "state": An Object that the individual game has serialized according to its
        own representation.
    "players": List of String. The uids of the players, where players[0] is the
        first player, players[1] is the second, etc.
    "moves": List of String. A list of moves in the game in chronological order.
        The encode/decoding to string is handled by the individual game code.
    "status": String (as defined in ChallengeStatus). Representing the game status
        (waiting for players, in progress, over, etc.)
    "cid": String. ID assigned to the game.
    "player_won": Int. If a player won, this is the index of the player that won
        Otherwise, it is -1.
}
*/

class Challenge {
  game_name = "";
  game_state = null;
  game_end = null;
  players = [];
  status = "";
  cid = "";
  moves = [];

  constructor(data) {
    makeObservable(this, {
      game_state: observable,
      players: observable,
      status: observable,
      game_end: observable,
      moves: observable,
      is_playing: computed,
      update_challenge_information: action,
    });
    this.cid = data.cid;
    this.game_name = data.game_name;
    this.game_state = new game_name_to_state_class[this.game_name](
      this,
      data.state.config
    );

    this.update_challenge_information(data);

    this.ViewComponent = game_name_to_view_component[this.game_name];
  }

  get is_playing() {
    return this.status == "IN_PROGRESS";
  }

  update_challenge_information(new_chall) {
    this.players = new_chall.players;
    this.status = new_chall.status;
    this.game_end = new_chall.player_won;
    this.moves = new_chall.moves;
    this.game_state.update_game_information(new_chall.state);
  }

  turn_of_user = (user_id) => {
    // 0-indexed player turn (ie. 1st player = 0, 2nd player = 1, ...)
    // -1 if not a player
    return this.players.findIndex((player_id) => player_id === user_id);
  };
}

export { Challenge };
