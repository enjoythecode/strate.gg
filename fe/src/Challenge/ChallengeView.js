import { observer } from "mobx-react";
import { toJS } from "mobx";
import React from "react";

import { useRootStore } from "../Store/RootStore.js";

import ChallengeDashboard from "./Components/ChallengeDashboard";

const ChallengeView = observer(({ challenge, move_handler }) => {
  const RootStore = useRootStore();
  let return_content;

  if (challenge == null) {
    return_content = "Loading the game!";
  } else {
    let game_is_in_progress = challenge.status === "IN_PROGRESS";
    let is_users_turn =
      challenge.game_state.turn ===
      challenge.turn_of_user(RootStore.client_uid);
    let player_can_move = game_is_in_progress && is_users_turn;

    return_content = (
      <div className="challenge-wrapper">
        <div className="challenge-board">
          <challenge.ViewComponent
            game_state={challenge.game_state}
            handle_move={player_can_move ? move_handler : undefined}
            last_move={
              challenge.moves.length > 0
                ? toJS(challenge.moves[challenge.moves.length - 1])
                : undefined
            }
          />
        </div>
        <ChallengeDashboard challenge={challenge} />
      </div>
    );
  }

  return <div data-testid="challenge">{return_content}</div>;
});

export { ChallengeView };
