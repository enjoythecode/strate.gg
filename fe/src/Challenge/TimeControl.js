import { makeObservable, observable, action, computed } from "mobx";

class TimeControl {
  config = { base_s: 0, increment_s: 0 };
  move_timestamps = [];

  constructor(data) {
    makeObservable(this, {
      config: observable,
      move_timestamps: observable,
      update_time_control_data: action,
    });

    this.update_time_control_data(data);
  }
  update_time_control_data(data) {
    if (data !== undefined && data !== {}) {
      this.config = data.time_config;
      this.move_timestamps = data.move_timestamps;
    }
  }
  total_time_used_of_player(player) {
    if (this.move_timestamps === undefined) {
      return 0;
    }

    let used_time = 0;
    let first_ts_i = player == 0 ? 2 : 1;

    for (let i = first_ts_i; i < this.move_timestamps.length; i += 2) {
      used_time += this.move_timestamps[i] - this.move_timestamps[i - 1];
    }

    return used_time;
  }
}

export default TimeControl;
