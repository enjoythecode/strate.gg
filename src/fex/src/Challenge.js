import { makeObservable, observable, computed, action, flow } from "mobx"
import { observer } from "mobx-react"
import React from "react"
//import { AbstractGameState } from "./AbstractGameState"

class Challenge {
    game_name = ""
    game_state = ""
    players = []
    turn = 0
    status = ""
    resolution = ""
    
    constructor(game_name){
        makeObservable(this, {
            game_state: observable,
            players: observable,
            turn: observable,
            status: observable,
            resolution: observable,

            //sckt_move_update: action,
            //sckt_meta_update: action,
        })
        this.game_name = game_name
        console.log("testing", game_name)
    }
}

const ChallengeView = observer(({ state }) =>(
    <div>
        <h1>Hello, world!</h1>
        <div style={{position:'absolute', top:'0', right:'0'}}>{state.active_users} online users!</div>
    </div>
))

const chall = new Challenge("amazons") 
//const ChallengeApp = () => (<ChallengeView Challenge={chall} />)

export default ChallengeView