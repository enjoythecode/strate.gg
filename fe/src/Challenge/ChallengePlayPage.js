import { observer } from "mobx-react";
import React, { useEffect } from "react";
import { useRootStore } from "Store/RootStore.js";
import { ChallengeView } from "Challenge/ChallengeView.js";

function get_game_id_from_url() {
  let params = new URLSearchParams(window.location.search);
  return params.get("cid");
}

const ChallengePlayPage = observer(() => {
  let cid = get_game_id_from_url();
  const RootStore = useRootStore();

  useEffect(() => {
    RootStore.socket.challenge_subscribe(cid);

    return function cleanup() {
      RootStore.socket.challenge_unsubscribe(cid);
    };
  }, [cid]);

  const challenge = RootStore.challenges.get(cid);
  if (challenge == null) {
    return <span>Loading the game!</span>;
  }

  return (
    <ChallengeView
      challenge={challenge}
      move_handler={(move) =>
        RootStore.socket.challenge_send_move({ cid: cid, move: move })
      }
    />
  );
});

export default ChallengePlayPage;
