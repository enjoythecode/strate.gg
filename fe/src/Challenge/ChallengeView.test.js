import { ChallengeView } from "./ChallengeView";
import { render, screen } from "@testing-library/react";
import { Challenge } from "./Challenge";
import { NEW_AMAZONS_CHALLENGE_DATA } from "./Challenge.test";

describe("ChallengeView", () => {
  it("renders text when it does not have challenge data", () => {
    render(<ChallengeView></ChallengeView>);
    let chall_view = screen.getByTestId("challenge");

    expect(chall_view.textContent).toBe("Loading the game!");
  });

  it("renders challenge with challenge data", () => {
    const challenge = new Challenge(NEW_AMAZONS_CHALLENGE_DATA);

    const { asFragment } = render(
      <ChallengeView challenge={challenge}></ChallengeView>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
