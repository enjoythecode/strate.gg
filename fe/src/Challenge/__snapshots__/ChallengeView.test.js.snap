// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChallengeView renders challenge with at least one move and the last move indicator 1`] = `
<DocumentFragment>
  <div
    data-testid="challenge"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-6s1xft-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-7 css-1a4ufwv-MuiGrid-root"
      >
        <div
          class="amazonsBoard"
          style="grid-template-columns: repeat(10, 1fr); grid-template-rows: repeat(10, 1fr);"
        >
          <div
            class="cellDark"
          />
          <div
            class="cellLight"
          />
          <div
            class="cellDark"
          />
          <div
            class="cellLight"
          />
          <div
            class="cellDark"
          />
          <div
            class="cellLight"
          />
          <div
            class="cellDark"
          />
          <div
            class="cellLight"
          />
          <div
            class="cellDark"
          />
          <div
            class="cellLight"
          />
          <div
            class="cellLight"
          />
          <div
            class="cellDark"
          />
          <div
            class="cellLight"
          />
          <div
            class="indicatorLastMove indicatorInner cellDark"
          />
          <div
            class="cellLight"
          />
          <div
            class="cellDark"
          />
          <div
            class="indicatorLastMove indicatorInner cellLight"
          />
          <div
            class="cellDark"
          />
          <div
            class="cellLight"
          />
          <div
            class="cellDark"
          />
          <div
            class="cellDark"
          />
          <div
            class="cellLight"
          />
          <div
            class="cellDark"
          />
          <div
            class="cellLight"
          />
          <div
            class="cellDark"
          />
          <div
            class="cellLight"
          />
          <div
            class="cellDark"
          />
          <div
            class="cellLight"
          />
          <div
            class="cellDark"
          />
          <div
            class="cellLight"
          />
          <div
            class="cellLight"
          />
          <div
            class="cellDark"
          />
          <div
            class="cellLight"
          />
          <div
            class="cellDark"
          />
          <div
            class="cellLight"
          />
          <div
            class="cellDark"
          />
          <div
            class="cellLight"
          />
          <div
            class="cellDark"
          />
          <div
            class="cellLight"
          />
          <div
            class="cellDark"
          />
          <div
            class="cellDark"
          />
          <div
            class="cellLight"
          />
          <div
            class="cellDark"
          />
          <div
            class="cellLight"
          />
          <div
            class="cellDark"
          />
          <div
            class="cellLight"
          />
          <div
            class="cellDark"
          />
          <div
            class="cellLight"
          />
          <div
            class="cellDark"
          />
          <div
            class="cellLight"
          />
          <div
            class="cellLight"
          />
          <div
            class="cellDark"
          />
          <div
            class="cellLight"
          />
          <div
            class="cellDark"
          />
          <div
            class="cellLight"
          />
          <div
            class="cellDark"
          />
          <div
            class="cellLight"
          />
          <div
            class="cellDark"
          />
          <div
            class="cellLight"
          />
          <div
            class="cellDark"
          />
          <div
            class="cellDark"
          />
          <div
            class="cellLight"
          />
          <div
            class="cellDark"
          />
          <div
            class="cellLight"
          />
          <div
            class="cellDark"
          />
          <div
            class="cellLight"
          />
          <div
            class="cellDark"
          />
          <div
            class="cellLight"
          />
          <div
            class="cellDark"
          />
          <div
            class="cellLight"
          />
          <div
            class="cellLight"
          />
          <div
            class="cellDark"
          />
          <div
            class="cellLight"
          />
          <div
            class="cellDark"
          />
          <div
            class="cellLight"
          />
          <div
            class="cellDark"
          />
          <div
            class="cellLight"
          />
          <div
            class="cellDark"
          />
          <div
            class="cellLight"
          />
          <div
            class="cellDark"
          />
          <div
            class="cellDark"
          />
          <div
            class="cellLight"
          />
          <div
            class="cellDark"
          />
          <div
            class="cellLight"
          />
          <div
            class="cellDark"
          />
          <div
            class="cellLight"
          />
          <div
            class="cellDark"
          />
          <div
            class="cellLight"
          />
          <div
            class="cellDark"
          />
          <div
            class="cellLight"
          />
          <div
            class="cellLight"
          />
          <div
            class="cellDark"
          />
          <div
            class="cellLight"
          />
          <div
            class="cellDark"
          />
          <div
            class="cellLight"
          />
          <div
            class="cellDark"
          />
          <div
            class="indicatorLastMove indicatorInner cellLight"
          />
          <div
            class="cellDark"
          />
          <div
            class="cellLight"
          />
          <div
            class="cellDark"
          />
          <img
            alt="Black Queen"
            class="amazonsPiece"
            src="/images/bqueen.png"
            style="left: 30%; top: 0%;"
          />
          <img
            alt="Black Queen"
            class="amazonsPiece"
            src="/images/bqueen.png"
            style="left: 60%; top: 0%;"
          />
          <img
            alt="Burnt Off Square"
            class="amazonsPiece"
            src="/images/fire.png"
            style="left: 30%; top: 10%;"
          />
          <img
            alt="White Queen"
            class="amazonsPiece"
            src="/images/wqueen.png"
            style="left: 60%; top: 10%;"
          />
          <img
            alt="Black Queen"
            class="amazonsPiece"
            src="/images/bqueen.png"
            style="left: 0%; top: 30%;"
          />
          <img
            alt="Black Queen"
            class="amazonsPiece"
            src="/images/bqueen.png"
            style="left: 90%; top: 30%;"
          />
          <img
            alt="White Queen"
            class="amazonsPiece"
            src="/images/wqueen.png"
            style="left: 0%; top: 60%;"
          />
          <img
            alt="White Queen"
            class="amazonsPiece"
            src="/images/wqueen.png"
            style="left: 90%; top: 60%;"
          />
          <img
            alt="White Queen"
            class="amazonsPiece"
            src="/images/wqueen.png"
            style="left: 30%; top: 90%;"
          />
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-5 css-urj5pv-MuiGrid-root"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-i0bk1x-MuiPaper-root"
        >
          <div
            class="MuiBox-root css-0"
          >
            <div>
              <ul
                class="MuiList-root MuiList-padding MuiList-dense css-190n62h-MuiList-root"
              >
                <li
                  class="MuiListItem-root MuiListItem-dense MuiListItem-gutters MuiListItem-padding css-6lnq02-MuiListItem-root"
                >
                  <div
                    class="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
                  >
                    <div
                      class="MuiAvatar-root MuiAvatar-circular css-1wlk0hk-MuiAvatar-root"
                    >
                      <img
                        alt="Black Queen Piece"
                        class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
                        src="/images/bqueen.png"
                      />
                    </div>
                  </div>
                  <div
                    class="MuiListItemText-root MuiListItemText-dense css-tlelie-MuiListItemText-root"
                  >
                    <p
                      class="MuiTypography-root MuiTypography-body1 css-maacv0-MuiTypography-root"
                    >
                      Guest #zochGd8
                    </p>
                  </div>
                </li>
              </ul>
            </div>
          </div>
          <hr
            class="MuiDivider-root MuiDivider-fullWidth css-1b008ma-MuiDivider-root"
          />
          <div
            style="max-height: 11em; overflow: auto; display: flex; flex-direction: column-reverse; font-family: monospace; padding: 1em;"
          >
            <ul
              style="list-style: none;"
            >
              <li>
                <p
                  class="MuiTypography-root MuiTypography-body1 css-an8rnp-MuiTypography-root"
                >
                  1. G10-G2/D2
                </p>
              </li>
            </ul>
          </div>
          <div
            class="MuiBox-root css-xi606m"
          >
            <em />
            <br />
            <em />
          </div>
          <hr
            class="MuiDivider-root MuiDivider-fullWidth css-1b008ma-MuiDivider-root"
          />
          <div
            class="MuiBox-root css-0"
          >
            <div>
              <ul
                class="MuiList-root MuiList-padding MuiList-dense css-h4y409-MuiList-root"
              >
                <li
                  class="MuiListItem-root MuiListItem-dense MuiListItem-gutters MuiListItem-padding css-6lnq02-MuiListItem-root"
                >
                  <div
                    class="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
                  >
                    <div
                      class="MuiAvatar-root MuiAvatar-circular css-1wlk0hk-MuiAvatar-root"
                    >
                      <img
                        alt="White Queen Piece"
                        class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
                        src="/images/wqueen.png"
                      />
                    </div>
                  </div>
                  <div
                    class="MuiListItemText-root MuiListItemText-dense css-tlelie-MuiListItemText-root"
                  >
                    <p
                      class="MuiTypography-root MuiTypography-body1 css-maacv0-MuiTypography-root"
                    >
                      Guest #T6A8IVn
                    </p>
                  </div>
                </li>
              </ul>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`ChallengeView renders challenge with challenge data 1`] = `
<DocumentFragment>
  <div
    data-testid="challenge"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-6s1xft-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-7 css-1a4ufwv-MuiGrid-root"
      >
        <div
          class="amazonsBoard"
          style="grid-template-columns: repeat(10, 1fr); grid-template-rows: repeat(10, 1fr);"
        >
          <div
            class="cellDark"
          />
          <div
            class="cellLight"
          />
          <div
            class="cellDark"
          />
          <div
            class="cellLight"
          />
          <div
            class="cellDark"
          />
          <div
            class="cellLight"
          />
          <div
            class="cellDark"
          />
          <div
            class="cellLight"
          />
          <div
            class="cellDark"
          />
          <div
            class="cellLight"
          />
          <div
            class="cellLight"
          />
          <div
            class="cellDark"
          />
          <div
            class="cellLight"
          />
          <div
            class="cellDark"
          />
          <div
            class="cellLight"
          />
          <div
            class="cellDark"
          />
          <div
            class="cellLight"
          />
          <div
            class="cellDark"
          />
          <div
            class="cellLight"
          />
          <div
            class="cellDark"
          />
          <div
            class="cellDark"
          />
          <div
            class="cellLight"
          />
          <div
            class="cellDark"
          />
          <div
            class="cellLight"
          />
          <div
            class="cellDark"
          />
          <div
            class="cellLight"
          />
          <div
            class="cellDark"
          />
          <div
            class="cellLight"
          />
          <div
            class="cellDark"
          />
          <div
            class="cellLight"
          />
          <div
            class="cellLight"
          />
          <div
            class="cellDark"
          />
          <div
            class="cellLight"
          />
          <div
            class="cellDark"
          />
          <div
            class="cellLight"
          />
          <div
            class="cellDark"
          />
          <div
            class="cellLight"
          />
          <div
            class="cellDark"
          />
          <div
            class="cellLight"
          />
          <div
            class="cellDark"
          />
          <div
            class="cellDark"
          />
          <div
            class="cellLight"
          />
          <div
            class="cellDark"
          />
          <div
            class="cellLight"
          />
          <div
            class="cellDark"
          />
          <div
            class="cellLight"
          />
          <div
            class="cellDark"
          />
          <div
            class="cellLight"
          />
          <div
            class="cellDark"
          />
          <div
            class="cellLight"
          />
          <div
            class="cellLight"
          />
          <div
            class="cellDark"
          />
          <div
            class="cellLight"
          />
          <div
            class="cellDark"
          />
          <div
            class="cellLight"
          />
          <div
            class="cellDark"
          />
          <div
            class="cellLight"
          />
          <div
            class="cellDark"
          />
          <div
            class="cellLight"
          />
          <div
            class="cellDark"
          />
          <div
            class="cellDark"
          />
          <div
            class="cellLight"
          />
          <div
            class="cellDark"
          />
          <div
            class="cellLight"
          />
          <div
            class="cellDark"
          />
          <div
            class="cellLight"
          />
          <div
            class="cellDark"
          />
          <div
            class="cellLight"
          />
          <div
            class="cellDark"
          />
          <div
            class="cellLight"
          />
          <div
            class="cellLight"
          />
          <div
            class="cellDark"
          />
          <div
            class="cellLight"
          />
          <div
            class="cellDark"
          />
          <div
            class="cellLight"
          />
          <div
            class="cellDark"
          />
          <div
            class="cellLight"
          />
          <div
            class="cellDark"
          />
          <div
            class="cellLight"
          />
          <div
            class="cellDark"
          />
          <div
            class="cellDark"
          />
          <div
            class="cellLight"
          />
          <div
            class="cellDark"
          />
          <div
            class="cellLight"
          />
          <div
            class="cellDark"
          />
          <div
            class="cellLight"
          />
          <div
            class="cellDark"
          />
          <div
            class="cellLight"
          />
          <div
            class="cellDark"
          />
          <div
            class="cellLight"
          />
          <div
            class="cellLight"
          />
          <div
            class="cellDark"
          />
          <div
            class="cellLight"
          />
          <div
            class="cellDark"
          />
          <div
            class="cellLight"
          />
          <div
            class="cellDark"
          />
          <div
            class="cellLight"
          />
          <div
            class="cellDark"
          />
          <div
            class="cellLight"
          />
          <div
            class="cellDark"
          />
          <img
            alt="Black Queen"
            class="amazonsPiece"
            src="/images/bqueen.png"
            style="left: 30%; top: 0%;"
          />
          <img
            alt="Black Queen"
            class="amazonsPiece"
            src="/images/bqueen.png"
            style="left: 60%; top: 0%;"
          />
          <img
            alt="Black Queen"
            class="amazonsPiece"
            src="/images/bqueen.png"
            style="left: 0%; top: 30%;"
          />
          <img
            alt="Black Queen"
            class="amazonsPiece"
            src="/images/bqueen.png"
            style="left: 90%; top: 30%;"
          />
          <img
            alt="White Queen"
            class="amazonsPiece"
            src="/images/wqueen.png"
            style="left: 0%; top: 60%;"
          />
          <img
            alt="White Queen"
            class="amazonsPiece"
            src="/images/wqueen.png"
            style="left: 90%; top: 60%;"
          />
          <img
            alt="White Queen"
            class="amazonsPiece"
            src="/images/wqueen.png"
            style="left: 30%; top: 90%;"
          />
          <img
            alt="White Queen"
            class="amazonsPiece"
            src="/images/wqueen.png"
            style="left: 60%; top: 90%;"
          />
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-5 css-urj5pv-MuiGrid-root"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-i0bk1x-MuiPaper-root"
        >
          <div
            class="MuiBox-root css-0"
          >
            <div>
              <ul
                class="MuiList-root MuiList-padding MuiList-dense css-h4y409-MuiList-root"
              >
                <li
                  class="MuiListItem-root MuiListItem-dense MuiListItem-gutters MuiListItem-padding css-6lnq02-MuiListItem-root"
                >
                  <div
                    class="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
                  >
                    <div
                      class="MuiAvatar-root MuiAvatar-circular css-1wlk0hk-MuiAvatar-root"
                    >
                      <img
                        alt="Black Queen Piece"
                        class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
                        src="/images/bqueen.png"
                      />
                    </div>
                  </div>
                  <div
                    class="MuiListItemText-root MuiListItemText-dense css-tlelie-MuiListItemText-root"
                  >
                    <button
                      class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-sghohy-MuiButtonBase-root-MuiButton-root"
                      tabindex="0"
                      type="button"
                    >
                      Join game as Black
                      <span
                        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                      />
                    </button>
                  </div>
                </li>
              </ul>
            </div>
          </div>
          <hr
            class="MuiDivider-root MuiDivider-fullWidth css-1b008ma-MuiDivider-root"
          />
          <div
            style="max-height: 11em; overflow: auto; display: flex; flex-direction: column-reverse; font-family: monospace; padding: 1em;"
          >
            <ul
              style="list-style: none;"
            />
          </div>
          <div
            class="MuiBox-root css-xi606m"
          >
            <em>
              Waiting for players to join...
            </em>
            <br />
            <em />
          </div>
          <hr
            class="MuiDivider-root MuiDivider-fullWidth css-1b008ma-MuiDivider-root"
          />
          <div
            class="MuiBox-root css-0"
          >
            <div>
              <ul
                class="MuiList-root MuiList-padding MuiList-dense css-h4y409-MuiList-root"
              >
                <li
                  class="MuiListItem-root MuiListItem-dense MuiListItem-gutters MuiListItem-padding css-6lnq02-MuiListItem-root"
                >
                  <div
                    class="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
                  >
                    <div
                      class="MuiAvatar-root MuiAvatar-circular css-1wlk0hk-MuiAvatar-root"
                    >
                      <img
                        alt="White Queen Piece"
                        class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
                        src="/images/wqueen.png"
                      />
                    </div>
                  </div>
                  <div
                    class="MuiListItemText-root MuiListItemText-dense css-tlelie-MuiListItemText-root"
                  >
                    <p
                      class="MuiTypography-root MuiTypography-body1 css-maacv0-MuiTypography-root"
                    >
                      Guest #T6A8IVn
                    </p>
                  </div>
                </li>
              </ul>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
